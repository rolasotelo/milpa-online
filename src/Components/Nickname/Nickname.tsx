import React from "react";

interface Props {}

const Nickname = (props: Props) => {
  return (
    <div className="max-w-md w-1/4 mx-auto flex justify-around bg-mexicanGreen-light p-3">
      <div className="self-center">
        <a href="/play">Ahoj</a>
      </div>
      <div>
        <input
          className="placeholder-gray-500 placeholder-opacity-100 ..."
          placeholder="Miroslava"
        />
      </div>
    </div>
  );
};

export default Nickname;
