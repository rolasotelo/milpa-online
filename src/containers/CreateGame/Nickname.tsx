import React from "react";
import { useTranslation } from "react-i18next";

interface Props {
  nickname: string;
  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
}

function Nickname(props: Props) {
  const { nickname, onChange } = props;
  const { t } = useTranslation();
  return (
    <div className='overflow-x-hidden' >
    <div
      className=" w-20.38rem md:w-49.81rem h-4.38rem mx-auto flex justify-around bg-green-banner-mobile md:bg-green-banner md:px-5 pt-6 mt-2 text-2xl md:text-3xl"
      style={{
        fontFamily: "goodlife-sans-condensed, sans-serif",
        fontWeight: 400,
        fontStyle: "normal",

      }}
    >
      <div
        className=" md:ml-6 text-milpaBeige-default"

      >
        <a href="/play">{t("play.greeting").toUpperCase()}</a>
      </div>
      <div className="md:ml-2">
        <input
          className=" placeholder-gray-500 placeholder-opacity-100 focus:outline-none bg-transparent text-milpaBlue-dark"
          placeholder="Gabinka?"
          value={nickname}
          onChange={(event) => {
            onChange(event);
          }}
        />
      </div>
    </div>
    </div>
  );
}

export default Nickname;
