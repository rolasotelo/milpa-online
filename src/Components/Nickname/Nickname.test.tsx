import React from "react";
import renderer from "react-test-renderer";

import Nickname from "./Nickname";

describe("<Nickname />", () => {
  const defaultProps = {
    nickname: "Gabinka",
    onChange: (event: React.ChangeEvent<HTMLInputElement>) => {},
    greeting: "HOLA",
  };
  const wrapper = renderer.create(<Nickname {...defaultProps} />);

  test("render", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
