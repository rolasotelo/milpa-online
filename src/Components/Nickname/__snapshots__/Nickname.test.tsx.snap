// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nickname /> render 1`] = `
<div
  className="w-20.38rem md:w-49.81rem h-4.38rem mx-auto flex justify-around bg-green-banner-mobile md:bg-green-banner p-2"
  style={
    Object {
      "fontFamily": "goodlife-sans-condensed, sans-serif",
      "fontSize": "2rem",
      "fontStyle": "normal",
      "fontWeight": 400,
    }
  }
>
  <div
    className="self-center mb-2 ml-6"
    style={
      Object {
        "color": "white",
      }
    }
  >
    <a
      href="/play"
    >
      HOLA
    </a>
  </div>
  <div
    className="ml-2"
  >
    <input
      className=" placeholder-gray-500 placeholder-opacity-100 focus:outline-none bg-transparent"
      onChange={[Function]}
      placeholder="Gabinka?"
      value="Gabinka"
    />
  </div>
</div>
`;
